File: app/src/main/java/com/martinlaizg/geofind/data/access/database/converter/DateTypeConverter.java
package com.martinlaizg.geofind.data.access.database.converter;

import androidx.room.TypeConverter;

import java.sql.Date;

public class DateTypeConverter {
	@TypeConverter
	public static Date toDate(Long dateLong) {
		return dateLong == null ?
				null :
				new Date(dateLong);
	}

	@TypeConverter
	public static Long fromDate(Date date) {
		return date == null ?
				null :
				date.getTime();
	}
}

File: app/src/main/java/com/martinlaizg/geofind/data/access/database/converter/DateTypeConverter.kt
package com.martinlaizg.geofind.data.access.database.converter

import androidx.room.TypeConverter
import java.sql.Date

object DateTypeConverter {
    @kotlin.jvm.JvmStatic
    @TypeConverter
    fun toDate(dateLong: Long?): Date? {
        return dateLong?.let { Date(it) }
    }

    @kotlin.jvm.JvmStatic
    @TypeConverter
    fun fromDate(date: Date?): Long? {
        return date?.time
    }
}